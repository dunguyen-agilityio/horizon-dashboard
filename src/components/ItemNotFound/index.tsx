// Components
import { Link } from '@nextui-org/link';
import { Text } from '@/components';

// Types
import { TEXT_SIZE } from '@/types/text';

interface IItemNotFoundProps {
  title: string;
  description: string;
  href?: string;
  label?: string;
}

const ItemNotFound = ({
  title,
  description,
  href,
  label = 'Explore Item',
}: IItemNotFoundProps) => (
  <div className="h-full flex items-center flex-col justify-center text-center">
    <Text
      as="h1"
      size={TEXT_SIZE['2xl']}
      className="mb-4 dark:text-white text-indigo-dark"
    >
      {title}
    </Text>
    <p className="text-lg text-gray-600 mb-8">{description}</p>
    {href && (
      <Link href={href}>
        <p className="inline-block bg-blue-450 dark:bg-purple-750 text-white py-3 px-6 rounded-full text-lg font-semibold shadow-lg hover:bg-blue-600">
          {label}
        </p>
      </Link>
    )}
  </div>
);

export default ItemNotFound;
