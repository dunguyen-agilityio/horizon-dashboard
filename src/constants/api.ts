export const API_ENTITY = {
  COMPLEX_REPORT: 'complex-report',
  CHECK_REPORT: 'check-report',
  FINANCIAL_REPORT: 'financial-report',
  USERS: 'users',
  SIGN_IN: 'auth/local',
  SIGN_UP: 'auth/local/register',
  FORGET_PASSWORD: 'auth/forgot-password',
  RESET_PASSWORD: 'auth/reset-password',
  CONTRIBUTOR: 'contributors',
};
