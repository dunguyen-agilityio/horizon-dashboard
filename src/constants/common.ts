export const QUERY_KEY = 'q';
export const STEP_KEY = 'step';
