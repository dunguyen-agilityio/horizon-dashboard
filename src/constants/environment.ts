export const API_ENDPOINT = process.env.API_ENDPOINT!;

// Firebase
export const FIREBASE_API_KEY = process.env.NEXT_PUBLIC_FIREBASE_API_KEY;
export const FIREBASE_AUTH_DOMAIN =
  process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN;
export const FIREBASE_PROJECT_ID = process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID;
export const FIREBASE_STORAGE_BUCKET =
  process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET;
export const FIREBASE_SENDER_ID = process.env.NEXT_PUBLIC_FIREBASE_SENDER_ID;
export const FIREBASE_APP_ID = process.env.NEXT_PUBLIC_FIREBASE_APP_ID;
export const FIREBASE_MEASUREMENT =
  process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT;
export const FIREBASE_CLOUD_KEY = process.env.FIREBASE_CLOUD_KEY;
