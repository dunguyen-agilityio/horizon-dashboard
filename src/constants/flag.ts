export const FLAG = {
  'United States': '/usa.webp',
  VietNam: '/vn.webp',
};

export const FLAG_KEYS = Object.keys(FLAG) as (keyof typeof FLAG)[];
