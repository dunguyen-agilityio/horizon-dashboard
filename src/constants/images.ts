export const STARBUCKS_IMAGE = {
  COVER: '/starbucks-cover.webp',
  LOGO: '/starbucks-logo.webp',
};

export const NFT_IMAGES = {
  COVER: '/nft-cover.webp',
};

export const STARBUCKS_IMAGE = {
  COVER: '/starbucks-cover.webp',
  LOGO: '/starbucks-logo.webp',
};
