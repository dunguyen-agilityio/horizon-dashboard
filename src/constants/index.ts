export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
