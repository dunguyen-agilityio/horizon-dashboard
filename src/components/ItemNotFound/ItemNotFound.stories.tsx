import type { Meta, StoryObj } from '@storybook/react';

// Components
import ItemNotFound from '.';

// Constants
import { PUBLIC_ROUTES } from '@/constants';

const meta = {
  title: 'Components/ItemNotFound',
  component: ItemNotFound,
  tags: ['autodocs'],
} satisfies Meta<typeof ItemNotFound>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ItemNotFoundWithoutNavigation: Story = {
  args: {
    title: 'Start Find Your Favorite NFTs',
    description:
      'Browse our collection and click the heart icon to add NFTs to your favorites list!',
  },
};

export const ItemNotFoundWithNavigation: Story = {
  args: {
    ...ItemNotFoundWithoutNavigation.args,
    href: PUBLIC_ROUTES.NFT_MARKETPLACE,
  },
};

export const ItemNotFoundWithLabel: Story = {
  args: {
    ...ItemNotFoundWithoutNavigation.args,
    href: PUBLIC_ROUTES.NFT_MARKETPLACE,
    label: 'Explore More NFT',
  },
};
