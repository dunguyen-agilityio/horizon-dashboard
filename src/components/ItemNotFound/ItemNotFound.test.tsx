import { fireEvent, render } from '@testing-library/react';

// Components
import ItemNotFound from '.';
import { PUBLIC_ROUTES } from '@/constants';

const title = 'Start Find Your Favorite NFTs';
const description =
  'Browse our collection and click the heart icon to add NFTs to your favorites list!';
const labelButton = 'Explore Products';

describe('ItemNotFound component', () => {
  const { container } = render(
    <ItemNotFound title={title} description={description} />,
  );
  it('Should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('Should render correct title and description', () => {
    const { getByText } = render(
      <ItemNotFound title={title} description={description} />,
    );
    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(description)).toBeInTheDocument();
  });

  it('Should render button to navigation', () => {
    const { getByText } = render(
      <ItemNotFound
        title={title}
        description={description}
        label={labelButton}
        href={PUBLIC_ROUTES.NFT_MARKETPLACE}
      />,
    );
    const buttonExplore = getByText(labelButton);
    expect(buttonExplore).toBeInTheDocument();
    fireEvent.click(buttonExplore);
  });
});
