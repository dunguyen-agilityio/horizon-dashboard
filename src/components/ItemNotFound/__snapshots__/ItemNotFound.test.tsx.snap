// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemNotFound component Should match snapshot 1`] = `
<div>
  <div
    class="h-full flex items-center flex-col justify-center text-center"
  >
    <h1
      class="text-primary dark:text-white text-2xl font-bold leading-2xl mb-4 dark:text-white text-indigo-dark"
    >
      Start Find Your Favorite NFTs
    </h1>
    <p
      class="text-lg text-gray-600 mb-8"
    >
      Browse our collection and click the heart icon to add NFTs to your favorites list!
    </p>
  </div>
</div>
`;
